### ISSUE TITLE
[question] test_requires in lockfile/SBOM

### ISSUE THREAD

**Issue by dkoerner-festo** [conan-io/conan-extensions#144]
### What is your question?

test_requires packages (e.g. gtest) are entered as "requires" in the lockfiles and are therefore also part of the SBOM.  I think such requires should not be included in the SBOM. Or is there any reason for this?

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
> In my opinion the test_requires and tool_requires should not part of the SBOM. To filter out these dependencies I would skip all item with "node.context == build" and "node.test == true". Would this be correct?

It would be good to have the feedback from experts, I am not sure what is the expected behavior, or maybe this should be an option for users?

**Comment by memsharded** [comment]
Maybe @jkowalleck or @hedtke have some feedback regarding this?

**Comment by hedtke** [comment]
Indeed, this detail was not considered before. They should either be ignored, or we add a new flag to add them. Do we have any favorite way here?

**Comment by dkoerner-festo** [comment]
I would prefer to ignore by default and add a flag to add them.

**Comment by dkoerner-festo** [comment]
https://github.com/conan-io/conan-extensions/pull/159 solves at least the build context part.

**Comment by memsharded** [comment]
Adding ``--no-test-requires`` should also be relatively easy, if someone wants to give it a try.

**Comment by hedtke** [comment]
I'm on sick leave until August 2025 and will not have the time to work on this

**Comment by memsharded** [comment]
Thanks for the feedback @hedtke, I'll try to remember not to ping you, I hope everything goes well in your leave. Thanks very much for everything!
